pyi.py

```diff
@@ -159,7 +159,40 @@ class TypeVarInfo(NamedTuple):
 )
 
 
+class PyflakesPreProcessor(ast.NodeTransformer):
+    """Transform AST prior to passing it to pyflakes.
+
+    This reduces false positives on recursive class definitions.
+    """
+
+    def visit_ClassDef(self, node: ast.ClassDef) -> ast.ClassDef:
+        self.generic_visit(node)
+        node.bases = [
+            # Remove the subscript to prevent F821 errors from being raised for (valid) recursive definitions.
+            base.value if isinstance(base, ast.Subscript) else base
+            for base in node.bases
+        ]
+        return node
+
+
 class PyiAwareFlakesChecker(FlakesChecker):
+    def __init__(self, tree: ast.AST, **kwargs: Any) -> None:
+        super().__init__(PyflakesPreProcessor().visit(tree), **kwargs)
```

Comment:
This should make it more resilient in case flake8 adds more arguments in the future or whatever:

```suggestion
    def __init__(self, tree: ast.AST, *args, **kwargs: Any) -> None:
        super().__init__(PyflakesPreProcessor().visit(tree), *args, **kwargs)
```